packages/util/src/SNARKHelpers.ts

```diff
@@ -13,5 +15,18 @@ export const babyJubNegativeOne = BigInt(
  */
 export function babyJubIsNegativeOne(value: string): boolean {
   const bigintValue = BigInt(value);
-  return bigintValue === babyJubNegativeOne || bigintValue === BigInt(-1);
+  return bigintValue === BABY_JUB_NEGATIVE_ONE || bigintValue === BigInt(-1);
+}
+/**
+ * Hashes a message to be signed with sha256 and truncates to fit into a
+ * baby jub jub field element.  The result includes the top 248 bits of
+ * the 256 bit hash.
+ *
+ * @param signal The initial message.
+ * @returns The outputted hash, fed in as a signal to the Semaphore proof.
+ */
+export function generateSnarkMessageHash(signal: string): bigint {
```

Comment:
Coincidentally I also added this function to the `util` package in the email PCD PR, which was just merged. You can remove this version or rename my one, so long as we don't end up duplicating it.